### PR TITLE
slingshot: check all the markdown references

### PR DESCRIPTION
**Pull request by oleganza**
This adds a script that checks that most references in the specs are valid:

1. Checks all the local anchors within each `.md` file.
2. Checks all cross-file references and anchors between `.md` files.
3. Checks if remote URLs are loadable (pings `http(s)://` links).
4. Travis checks these on each PR.
5. `/vendor/` paths are ignored.

### Limitations

1. The title-to-anchor convertor is ad-hoc and may not produce the same result as Github/Markdown, so this may turn into false negatives or false positives.
2. Links to anchors to remote pages are not fully checked: only the URL is pinged, but the anchor within it is not verified (so it may become stale).
3. Remote URLs may die causing your PRs be rejected by Travis. In such case it's a good opportunity to fix the URLs or revisit the strictness of the script.

### CODE REVIEW

**Comment by oleganza** [comment]
Example:

image

**Comment by cathieyun** [comment]
This seems good! Do you know how much it time it adds to Travis's runtime? Just curious...

File: check-markdown-links.rb [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+          $check_links_failed = true
+        end
+      end
+    end
+  end
+end
+
+def check_url(url, cache = {})
+  return true if cache[url]
+  if url == "https://dx.doi.org/10.6028/NIST.FIPS.202"
```
Comment (cathieyun) [inline check-markdown-links.rb]: what's the reason to special-case this link?

Comment (oleganza) [reply check-markdown-links.rb]: sometimes it does not work if fetched from a script instead of a browser

**Pull request merged**